Storage/SQL/SQLiteHistoryRecommendations.swift

```diff
@@ -16,77 +16,57 @@ extension SQLiteHistory: HistoryRecommendations {
-        if blacklistedHosts.count > 0 {
-            blacklistSubquery = "SELECT " + "\(TableDomains).id" +
-                " FROM " + "\(TableDomains)" +
-                " WHERE " + "\(TableDomains).domain" + " IN " + BrowserDB.varlist(blacklistedHosts.count)
-        }
+        let blacklistSubquery = "SELECT \(TableDomains).id FROM \(TableDomains) " +
+                "WHERE " + "\(TableDomains).domain" + " IN " + BrowserDB.varlist(blacklistedHosts.count)
 
-        let removeMultipleDomainsSubquery =
-            "   INNER JOIN (SELECT \(ViewHistoryVisits).domain_id AS domain_id, MAX(\(ViewHistoryVisits).visitDate) AS visit_date" +
-            "   FROM \(ViewHistoryVisits)" +
-            "   GROUP BY \(ViewHistoryVisits).domain_id) AS domains ON domains.domain_id = \(TableHistory).domain_id AND visitDate = domains.visit_date"
+        /*
+         Lets get some history that we will then rank based on the metadata availible to us
+         When grouping by domain. Choose the url with the largest title and largest url.
+         We also want to filter out sites that appear in the top sites or are part of the blacklist.
+         From these 100 we will show less than a dozen.
+         */
 
-        let subQuerySiteProjection = "historyID, url, siteTitle, guid, visitCount, visitDate, is_bookmarked"
-        let nonRecentHistory =
-            "SELECT \(subQuerySiteProjection) FROM (" +
-            "   SELECT \(TableHistory).id as historyID, url, title AS siteTitle, guid, visitDate, \(TableHistory).domain_id," +
+        let newquery =
+            "SELECT historyID, url, title, guid, visitCount, visitDate, is_bookmarked " +
+            "FROM (" +
+            "   SELECT url, siteID as historyID, title, guid, visitDate," +
             "       (SELECT COUNT(1) FROM \(TableVisits) WHERE s = \(TableVisits).siteID) AS visitCount," +
             "       (SELECT COUNT(1) FROM \(ViewBookmarksLocalOnMirror) WHERE \(ViewBookmarksLocalOnMirror).bmkUri == url) AS is_bookmarked" +
-            "   FROM (" +
-            "       SELECT siteID AS s, max(date) AS visitDate" +
-            "       FROM \(TableVisits)" +
-            "       WHERE date < ?" +
-            "       GROUP BY siteID" +
-            "       ORDER BY visitDate DESC" +
-            "   )" +
-            "   LEFT JOIN \(TableHistory) ON \(TableHistory).id = s" +
-                removeMultipleDomainsSubquery +
-            "   WHERE visitCount <= 3 AND title NOT NULL AND title != '' AND is_bookmarked == 0 AND url NOT IN" +
-            "       (SELECT \(TableActivityStreamBlocklist).url FROM \(TableActivityStreamBlocklist))" +
-            "        AND \(TableHistory).domain_id NOT IN ("
-                    + blacklistSubquery + ")" +
-            "   LIMIT \(historyLimit)" +
-            ")"
+            "   FROM (SELECT  *, siteID AS s, min(date) AS visitDate, max(title), max(url) FROM visits  LEFT JOIN history ON history.id = s group by domain_id)" +
```

Comment:
max(title) and max(url) will select the url from that domain that has the largest url/title.